### PR TITLE
feat: Clear all previous notifications if a new notification is added to a conversation

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2771/clear-all-previous-notifications-if-a-new-notification-is-added-to-a

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -121,6 +121,9 @@ def process_notification_delivery
     # In future, we could probably add condition here to enqueue the job for 30 seconds later
     # when push enabled and then check in email job whether notification has been read already.
     Notification::EmailNotificationJob.perform_later(self)
+
+    # Remove duplicate notifications
+    Notification::RemoveDuplicateNotificationJob.perform_later(self)
```
Comment (sojan-official) [inline app/models/notification.rb:126]: lets add a spec to ensure this job is called on notification creation

**Review by sojan-official — COMMENTED** [review]
Looks good. minor comment

**Review by sojan-official — APPROVED** [review]
Lgtm

**Comment by muhsin-k** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **NoMethodError: undefined method `display_id' for #<Message id: 34580104, content: "lúc test trên aha.com thì nó ...** `Sidekiq/Migration::RemoveMessageNotifications` [View Issue](https://chatwoot-p3.sentry.io/issues/4702907738/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎

There were some notifications with a primary actor message. I have resolved this issue by manually clearing the notifications.

**Pull request merged**